FlowCrypt/Controllers/SideMenu/Menu/MyMenuViewController.swift

```diff
@@ -64,14 +65,21 @@ final class MyMenuViewController: ASDKViewController<ASDisplayNode> {
     private var isFirstLaunch = true
 
     init(
-        foldersProvider: FoldersServiceType = FoldersService(),
-        dataService: DataServiceType = DataService.shared,
+        appContext: AppContext,
+        foldersProvider: FoldersServiceType? = nil,
         globalRouter: GlobalRouterType = GlobalRouter(),
         decorator: MyMenuViewDecoratorType = MyMenuViewDecorator(),
         tableNode: ASTableNode = TableNode()
     ) {
-        self.foldersProvider = foldersProvider
-        self.dataService = dataService
+        guard let currentUser = appContext.dataService.currentUser else {
+            fatalError("no current user") // todo - dependency-inject
+        }
+        self.currentUser = currentUser
+        self.appContext = appContext
+        self.foldersProvider = foldersProvider ?? FoldersService(
+            encryptedStorage: appContext.encryptedStorage,
+            remoteFoldersProvider: appContext.getRequiredMailProvider().remoteFoldersProvider
+        )
```

Comment:
I initialized `FolderService` many times this way, good candidate to go into `AppContext` (a method, maybe, similar to `EmailProvider`)